src/tests/test_analysis.py

```diff
@@ -1,109 +1,78 @@
 """
 test_analysis.py
 
-Unit tests for EdiAnalyzer
+General test cases for the analysis package which tests message classification.
+Tests for specific formats are implemented within separate modules named test_<format>_analysis.py.
+For example: test_fhir_analysis.py, test_x12_analysis.py, etc
 """
-from edi.models import EdiMessageMetadata
-from edi.analysis import EdiAnalyzer
-from edi.models import EdiMessageType, BaseMessageType
 import pytest
+from edi.analysis import get_analyzer, FhirAnalyzer, Hl7Analyzer, X12Analyzer
+from edi.models import BaseMessageFormat, EdiMessageFormat
 
 
 @pytest.mark.parametrize(
-    "input_data",
-    [None, "", "        ", "John, Doe, MD, 1400 Anyhoo Lane"],
+    "input_message",
+    [None, "", "        ", "IS"],
 )
-def test_init_value_error(input_data):
+def test_analyze_invalid_message(input_message):
     with pytest.raises(ValueError):
-        EdiAnalyzer(input_message=input_data)
+        get_analyzer(input_message)
 
 
-@pytest.mark.parametrize(
-    "fixture_name, base_message_type, message_type",
-    [
-        ("hl7_message", BaseMessageType.TEXT, EdiMessageType.HL7),
-        ("x12_message", BaseMessageType.TEXT, EdiMessageType.X12),
-        ("fhir_json_message", BaseMessageType.JSON, EdiMessageType.FHIR),
-        ("fhir_xml_message", BaseMessageType.XML, EdiMessageType.FHIR),
-    ],
-)
-def test_init(fixture_name, base_message_type, message_type, request):
-    message_fixture = request.getfixturevalue(fixture_name)
-    analyzer = EdiAnalyzer(message_fixture)
-    assert analyzer.input_message == message_fixture
-    assert analyzer.base_message_type == base_message_type
-    assert analyzer.message_type == message_type
-
-
-def test_analyze_hl7(hl7_message):
-    expected_data = {
-        "baseMessageType": "TEXT",
-        "messageType": "HL7",
-        "specificationVersion": "2.6",
-        "checksum": "dce92fa2bb05ba55f975dcef9e9615d45e33981c36d46895f349886a87364d60",
-        "messageSize": 884,
-        "recordCount": 8,
-    }
-    analyzer = EdiAnalyzer(hl7_message)
-    actual_metadata = analyzer.analyze()
-
-    expected_metadata = EdiMessageMetadata(**expected_data)
-    assert actual_metadata.dict() == expected_metadata.dict()
-
-
-def test_analyze_x12(x12_message):
-    expected_data = {
-        "baseMessageType": "TEXT",
-        "messageType": "X12",
-        "specificationVersion": "005010X279A1",
-        "checksum": "d7a928f396efa0bb15277991bd8d4d9a2506d751f9de8b344c1a3e5f8c45a409",
-        "messageSize": 509,
-        "recordCount": 17,
-    }
-    analyzer = EdiAnalyzer(x12_message)
-    actual_metadata = analyzer.analyze()
-
-    expected_metadata = EdiMessageMetadata(**expected_data)
-    assert actual_metadata.dict() == expected_metadata.dict()
+def test_analyze_fhir_json(fhir_json_message):
+    analyzer = get_analyzer(fhir_json_message)
+    assert isinstance(analyzer, FhirAnalyzer)
+
+    edi_message_metadata = analyzer.analyze()
```

Comment:
What if, instead of (or in addition to) the two step process to call get_analyzer() then analyzer.analyze(), there is a top-level method analyze() that returns the metadata?  Underneath, it would create the correct analyzer and call analyze for us.

Reply:
That makes sense. I'll replace the current approach with a single call. We can revisit if we have the need to split things out into two operations.